src/components/Brush/index.js

```diff
@@ -1,81 +1,207 @@
-// Not implemented yet. Should write our own brush component,
-// that's the last thing we're not handling ourselves - will mess up rendering cycle (i think)
-
 import React from 'react';
 import PropTypes from 'prop-types';
-import * as d3 from 'd3';
-import { createXScale } from '../../utils/scale-helpers';
 
 class Brush extends React.Component {
+  state = {
+    dragStartOverlay: null,
+    dragStartSelection: null,
+    isDraggingHandleEast: false,
+    isDraggingHandleWest: false,
+    isDraggingOverlay: false,
+    isDraggingSelection: false,
+  };
+
   componentDidMount() {
-    const { width, height } = this.props;
-    this.brush = d3
-      .brushX()
-      .extent([[0, 0], [width, height]])
-      .on('brush end', this.didBrush);
-    this.selection = d3.select(this.brushNode);
-    this.selection.call(this.brush);
+    window.addEventListener('mouseup', this.onMouseUp);
```

Comment:
Do you want to check `zoomable` before we register these?

Reply:
I don't think the cost of registering them (considering we're throwing early if it's not zoomable) is any high. No strong opinion.

Reply:
Yeah me neither. Leave it in.